docs/design-proposals/apiserver/openstack-api-support.md

```diff
@@ -0,0 +1,615 @@
+---
+title: Support Openstack Compute APIs in Arktos
+authors:
+  - "@yb01"
+---
+
+# Support Openstack Compute APIs in Arktos
+## Background and context
+To boost Arktos VM adoption and to let Openstack users to easily explore Arktos VM support without a lot of changes from 
+the users’ applications, it is decided to add a new set of APIs in Arktos to match Openstack Server related operations. 
+Besides better user experiences, this new set ofAPIs will also help existing Openstack perf tests to be relatively easily 
+converted to run against Arktos.
+
+## Goal and non-goals
+For the Arktos 130 release (1-30-2022):
+
+Goals:
+
+1. New set of REST interfaces in Arktos to support Openstack requests for Server operations
+1. New set of REST interfaces in Arktos to support Openstack requests for Server actions
+1. Initial Batch support with Arktos replicaset
+
+Non-goals:
+
+1. Nova or Nova test tool can switch to Arktos for basic VM operation, by switching to the Arktos URL. 
+1. Openstack client-side SDK or other dev tools are not in the scope of this design doc.
+1. Openstack API other than VM CRUD, VM Actions, is not in the scope of this design doc.
+
+## New APIs in Arktos
+### New API route paths:
+The new routes for servers
+
+* 	{Arktos-service-url}/servers
+* 	{Arktos-service-url}/servers/{server-id}/detail
+
+The new route for server actions
+
+* {Arktos-service-url}/servers/{server-id}/action
```

Comment:
It is unclear that how different actions will be differentiated in URL and request body

Reply:
this is Openstack specific. the action details, action, action extra parameters etc, are all in the request body. the URL is followed what OS has.

Reply:
this in the VM action section below.